src/components/Asana.tsx

```diff
@@ -1,37 +1,41 @@
 import React from 'react';
-import { Card, Row, Button } from 'antd';
+import { Card, Row, Button, Spin } from 'antd';
 import { map } from 'lodash';
-import { ITask } from './Task';
+
 import '../styles/Asana.css';
+
 import logo from '../asana_logo.png';
 import AsanaTask from './AsanaTask';
+import { createClient } from '../lib/asana';
+import { useAsanaTasks, useAsanaCredentials } from '../hooks/asana';
 
-export interface IAsanaProps {
-  project?: string
-}
-
-// TODO Fetch tasks
-const tasks: ITask[] = [{
-  description: 'Support search asana tasks from Command Palette',
-}, {
-  description: 'Add pagination to Asana integration',
-}]
+export interface IAsanaProps {}
 
 const Asana = (props: IAsanaProps) => {
+  const [credentials] = useAsanaCredentials();
+  const isAsanaAuthorized = !!credentials.access_token
+  const { tasks, workspace, loading } = useAsanaTasks()
   return (
     <Card className="Asana">
       <Row className="Asana-title">Asana</Row>
-      {props.project && <Row className="Asana-project">{props.project}</Row>}
-      {!props.project && <Row className="Asana-connect">
-        <Button className="Asana-connect-button">
+      {!isAsanaAuthorized && <Row className="Asana-connect">
+        <Button href={createClient().app.asanaAuthorizeUrl()} className="Asana-connect-button">
           <img src={logo} className="Asana-logo" height='32px' alt=''/> Connect Asana
         </Button>
       </Row>}
-      {map(tasks, task => (
-        <Row key={task.description}>
-          <AsanaTask {...task}/>
-        </Row>
-      ))}
+    {isAsanaAuthorized && (
+      <React.Fragment>
+        <Row className="Asana-project">{workspace}</Row>
+        {loading && <Row className="Asana-loading">
+          <Spin />
+        </Row>}
+        {map(tasks, task => (
```

Comment:
Is there any specific reason to use `lodash.map` vs `tasks.map` here?

Reply:
If for some reason `tasks` is `undefined` or `null` then `lodash.map` will happily accept it and return an empty list that will render into nothing where as `tasks.map` would cause and error. In this context `tasks` should not be `undefined` or `null` as I have ensured it is initialized and I have protected it with typescript typings but sometimes things happen at runtime that are not expected.

Reply:
Thanks for the answer!